project/Build.scala

```diff
@@ -583,6 +586,15 @@ object Build {
         (testOnly in Test).toTask(cmd)
       }.evaluated,
 
+      testFromTasty := Def.inputTaskDyn {
+        val args: Seq[String] = spaceDelimited("<arg>").parsed
+        val cmd = " dotty.tools.dotc.FromTastyTests -- " + {
+          if (args.nonEmpty) " -Ddotty.tests.filter=" + args.mkString(" ")
+          else ""
+        }
+        (testOnly in Test).toTask(cmd)
+      }.evaluated,
```

Comment:
Let's abstract over the testOnly input task:
```scala
def testOnly(test: String, options: String) = Def.inputTaskDyn {
  val args = spaceDelimited("<arg>").parsed
  val cmd = s" $test -- $options" + {
    if (args.nonEmpty) " -Ddotty.tests.filter=" + args.mkString(" ")
    else ""
  }
  (testOnly in Test).toTask(cmd)
}
```

Reply:
Done